src/main/java/com/teamcautionrobotics/robot/DriveBase.java

```diff
@@ -61,6 +75,46 @@ public void drive(double speed) {
         drive(speed, speed);
     }
 
+    public double getJerkLimit() {
+        return jerkLimit;
+    }
+
+    public void setJerkLimit(double jerkLimit) {
+        this.jerkLimit = jerkLimit;
+    }
+
+    public void driveSmoothly(double leftInput, double rightInput) {
+        updateDerivatives(leftInput, rightInput);
+        double leftPower = leftInput;
+        double rightPower = rightInput;
+
+        // limit jerk for each side if predicted jerk is too high
+        if (leftInputDerivative > jerkLimit) {
+            leftPower = limitJerk(leftInput, lastLeftPower);
+        }
+
+        if (rightInputDerivative > jerkLimit) {
+            rightPower = limitJerk(rightInput, lastRightPower);
+        }
+        drive(leftPower, rightPower);
+
+        lastLeftPower = leftPower;
+        lastRightPower = rightPower;
+    }
+
+    private void updateDerivatives(double leftInput, double rightInput) {
```

Comment:
I do not think this function is necessary, since the (left|right)InputDerivative is only used in `smoothDrive()`. I think it would also be cleaner to make `(left|right)InputDerivative` be local variables to `smoothDrive()`